core/src/banking_stage/consume_worker.rs

```diff
@@ -122,6 +148,460 @@ fn try_drain_iter<T>(work: T, receiver: &Receiver<T>) -> impl Iterator<Item = T>
     std::iter::once(work).chain(receiver.try_iter())
 }
 
+/// Metrics tracking number of packets processed by the consume worker.
+/// These are atomic, and intended to be reported by the scheduling thread
+/// since the consume worker thread is sleeping unless there is work to be
+/// done.
+pub(crate) struct ConsumeWorkerMetrics {
+    id: u32,
+    interval: AtomicInterval,
+    latch: AtomicBool,
+
+    count_metrics: ConsumeWorkerCountMetrics,
+    error_metrics: ConsumeWorkerTransactionErrorMetrics,
+    timing_metrics: ConsumeWorkerTimingMetrics,
+}
+
+impl ConsumeWorkerMetrics {
+    /// Report and reset metrics iff the interval has elapsed and the worker did some work.
+    pub fn maybe_report_and_reset(&self) {
+        const REPORT_INTERVAL_MS: u64 = 1000;
+        if self.interval.should_update(REPORT_INTERVAL_MS)
+            && self.latch.swap(false, Ordering::Relaxed)
```

Comment:
So the 3 metric kinds (count, error, timing) all share the same report interval and this `latch` variable. The `latch` is used to tell us if anything happened, i.e. the worker thread didn't just sleep the entire time. When worker receives work it sets the `latch` to true, and it will be reset to false (by the scheduler) on the first interval expiration after that.